### PR TITLE
feat: KPI-234 new investment date report

### PR DESCRIPTION
**Pull request by FreddyJR1995**
## Jira Link

[Go to Jira](https://kpinetwork.atlassian.net/browse/KPI-234)

## Types of changes

- [ ] Hotfix (Bug detected)
- [x] Feature (Issue from an Active Sprint)

## Checklist:

- [x] Unit Test.
- [x] Coverage.

### CODE REVIEW

**Review by kellycastrof — CHANGES_REQUESTED** [review]

File: src/service/investment_date_report/investment_date_report.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        company_data = dict()
+        filters = self.metric_repository.add_company_filters(**conditions)
+        data = self.by_metric_report.get_records(metric, years, filters)
+
+        profiles, sizes = self.by_metric_report.get_profiles(filters)
+
+        for id in data:
+            company = data[id]
+            if company["id"] == company_id:
+                if self.by_metric_report.is_in_range(profiles.get(id), **conditions):
```
[the remark this answers is not in the document]
Comment (FreddyJR1995) [reply src/service/investment_date_report/investment_date_report.py]: I have already changed

**Review by kellycastrof — APPROVED** [review]

File: infrastructure/network/gateways.tf [hunk 2]
```diff
@@ -354,6 +358,18 @@ resource "aws_api_gateway_method" "get_investment_year_report_method" {
...
 }
+resource "aws_api_gateway_method" "get_investment_date_report_method" {
+  rest_api_id   = aws_api_gateway_rest_api.api.id
+  resource_id   = aws_api_gateway_resource.investment_date_report.id
+  http_method   = "GET"
+  authorization = "CUSTOM"
+  authorizer_id = aws_api_gateway_authorizer.kpi_authorizer.id
+
+  request_parameters = {
+    "method.request.querystring.metrics" = false
```
Comment (kellycastrof) [inline infrastructure/network/gateways.tf:369]: There are also needed the filter options, see the other reports

File: src/handlers/investment_date_report/get_investment_date_report_handler.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        "Access-Control-Allow-Methods": "OPTIONS,POST,GET",
+    }
+
+
+def handler(event, _):
+    try:
+        report_service = get_investment_date_report_service()
+        user_id = get_user_id_from_event(event)
+        access = verify_user_access(user_id)
+        company_id = event.get("pathParameters").get("company_id")
```
Comment (kellycastrof) [inline src/handlers/investment_date_report/get_investment_date_report_handler.py:68]: We don't have the company_id  in the pathParameter (review infrastructure API gateway file {company_id}), so please take it from query string parameters or define the path in the infrastructure for the company_id

**Review by kellycastrof — CHANGES_REQUESTED** [review]

File: src/handlers/investment_date_report/get_investment_date_report_handler.py (hunk 3 above)
Comment (FreddyJR1995) [reply src/handlers/investment_date_report/get_investment_date_report_handler.py]: I added it in the queryStringParameters.

**Review by kellycastrof — APPROVED** [review]

**Pull request merged**